hs_app_timeseries/receivers.py

```diff
@@ -96,44 +133,48 @@ def post_create_resource_handler(sender, **kwargs):
             validate_err_message = _validate_odm2_db_file(fl_obj_name)
             if not validate_err_message:
                 extract_err_message = _extract_metadata(resource, fl_obj_name)
-                utils.resource_modified(resource, user)
                 if extract_err_message:
+                    # delete the invalid file
+                    delete_resource_file_only(resource, res_file)
+                    # cleanup any extracted metadata
+                    _delete_extracted_metadata(resource)
                     validate_files_dict['are_files_valid'] = False
-                    validate_files_dict['message'] = extract_err_message + " (Failed to extract all metadata)"
+                    extract_err_message += " " + FILE_UPLOAD_ERROR_MESSAGE
+                    validate_files_dict['message'] = extract_err_message
+                else:
+                    utils.resource_modified(resource, user)
             else:
+                # delete the invalid file
+                delete_resource_file_only(resource, res_file)
                 validate_files_dict['are_files_valid'] = False
-                validate_files_dict['message'] = validate_err_message + " (Metadata was not extracted)"
+                validate_err_message += " " + FILE_UPLOAD_ERROR_MESSAGE
+                validate_files_dict['message'] = validate_err_message
 
         if res_file.fed_resource_file_name_or_path and fl_obj_name:
             shutil.rmtree(os.path.dirname(fl_obj_name))
 
+
 @receiver(pre_metadata_element_create, sender=TimeSeriesResource)
 def metadata_element_pre_create_handler(sender, **kwargs):
     element_name = kwargs['element_name'].lower()
     request = kwargs['request']
-
-    if element_name == "site":
-        element_form = SiteValidationForm(request.POST)
-    elif element_name == 'variable':
-        element_form = VariableValidationForm(request.POST)
-    elif element_name == 'method':
-        element_form = MethodValidationForm(request.POST)
-    elif element_name == 'processinglevel':
-        element_form = ProcessingLevelValidationForm(request.POST)
-    elif element_name == 'timeseriesresult':
-        element_form = TimeSeriesResultValidationForm(request.POST)
-
-    if element_form.is_valid():
-        return {'is_valid': True, 'element_data_dict': element_form.cleaned_data}
-    else:
-        return {'is_valid': False, 'element_data_dict': None}
+    return _validate_metadata(request, element_name)
 
 
 @receiver(pre_metadata_element_update, sender=TimeSeriesResource)
 def metadata_element_pre_update_handler(sender, **kwargs):
     element_name = kwargs['element_name'].lower()
     request = kwargs['request']
+    return _validate_metadata(request, element_name)
+
+"""
+ Since each of the timeseries metadata element is required no need to listen to any delete signal
+ The timeseries landing page should not have delete UI functionality for the resource specific
+ metadata elements
+"""
+
```

Comment:
@pkdash this comment above is detached to any method, should it be added as a docstring to some method?